### ISSUE TITLE
Column header titles are not consistent with plurals elsewhere (and ignore custom plurals)

### ISSUE THREAD

**Issue by NoTuxNoBux** [keystonejs/keystone#7740]
Column header titles seem to have their own word splitting for words. For example, with a list `QRReferencePoint`:

![afbeelding](https://user-images.githubusercontent.com/73220431/180792056-f82bef1f-d4d1-4601-a8c2-bd1c1fcc13f7.png)

Notice `Qr Reference Points` in the header, and `QRReference Points` as generated plural in the cells.

A closely related problem is that they ignore any custom plurals. I've configured a custom `label`, `singular` and `plural` of "QR Reference Point" and "QR Reference Points" to properly format "QRReference", but the title seems to ignore it:

![afbeelding](https://user-images.githubusercontent.com/73220431/180791244-f5226f68-61d9-4eab-aea3-4b9487480ccc.png)

### Title Inconsistency

1. Have two linked entities in a one-to-many relationship where the inverse side of the relationship has a name containing multiple capital letters, e.g. A has one or more `QRReferencePoint`s.
2. Add the field referencing `QRReferencePoint`s (in A, meaning `A.QRReferencePoints`) to the initial columns of A.
3. Observe the mismatch in cell output and column title.

### Reproduce Ignored Plurals

1. Have two linked entities in a one-to-many relationship, A has one or more B's.
2. Give B a custom `label`, `singular`, and `plural`.
3. Add the field referencing B (in A, meaning `A.B`) to the initial columns of A.
4. Observe the mismatch in cell output and column title.

## Expected Behaviour

The column title uses the same plural as the cells do, both when plurals have been customized and when they don't.

## Information

* Node Version: 16
* Browser: Firefox 102

**Comment by dcousens** [comment]
@NoTuxNoBux to help us verify we have resolved your problem, can you provide an example schema?

**Comment by NoTuxNoBux** [comment]
The following schema reproduces the column title not using the plural configured for the entity:

```ts
import { list } from '@keystone-6/core';
import { password, relationship, text } from '@keystone-6/core/fields';

export const lists = {
    User: list({
        fields: {
            email: text({
                validation: { isRequired: true, length: { max: 100 } },
                isIndexed: 'unique',
                isFilterable: true,
            }),
            password: password(),
        },
    }),
    TestA: list({
        fields: {
            qrReferencePoints: relationship({
                ref: 'QRReferencePoint.testA',
                many: true,
            }),
        },
        ui: {
            listView: {
                initialColumns: ['id', 'qrReferencePoints'],
            },
        },
    }),
    QRReferencePoint: list({
        ui: {
            label: 'QR Reference Points',
            singular: 'QR Reference Point',
            plural: 'QR Reference Points',
            path: 'qr-reference-points',
        },
        fields: {
            testA: relationship({
                ref: 'TestA.qrReferencePoints',
                many: false,
            }),
        },
    }),
};
```

![afbeelding](https://user-images.githubusercontent.com/73220431/180946847-2d179f05-36dc-4b5b-8126-63fd09b32f6e.png)

**Comment by flexdinesh** [comment]
Which version of keystone are you using @NoTuxNoBux ? I checked the label config in a local codebase and can confirm it works the way it's intended to work in `"@keystone-ui/core": "^5.0.0"` `"@keystone-6/core": "^2.1.0"`

Eg. using this schema, we get the right label in table header and cell values.
```
TestA: list({
        fields: {
            qrReferencePoints: relationship({
                ref: 'QRReferencePoint.testA',
                many: true,
                ui: {
                  displayMode: 'count',
                },
            }),
        },
        ui: {
            listView: {
                initialColumns: ['id', 'qrReferencePoints'],
            },
        },
    }),
...
QRReferencePoint: list({
        ui: {
            label: 'QR Reference Points',
            singular: 'QR Reference Point',
            plural: 'QR Reference Points',
        },
        fields: {
            testA: relationship({
                ref: 'TestA.qrReferencePoints',
                many: false,
            }),
        },
    }),
```

**Comment by NoTuxNoBux** [comment]
I'm using the very latest version at the time of writing (14 July 2022). I also checked my lock file and it's using the same versions of the UI and core, and reinstalled all dependencies from a clean `node_modules` folder to ensure nothing got stuck whilst upgrading.

If I understood correctly, the column title you are seeing is actually `QR Reference Point`, with a capital `R` as specified by the `plural`, and not a lower-case `R`, as determined automatically by the table somehow, and shown in the screenshot?

**Comment by flexdinesh** [comment]
Yup. Here you go —

```
export const QRReferencePoint = list({
  ui: {
    label: 'QR Reference Point',
    singular: 'QR Reference Point',
    plural: 'QR Reference Points',
  },
  fields: {
    label: text(),
    description: text(),
    user: relationship({ ref: 'User.qrReferencePoints', many: false }),
  },
});
```

<img width="310" alt="nav" src="https://user-images.githubusercontent.com/5777880/180991551-4568c09a-a0b6-4104-9d10-4402d3aad336.png">
<img width="308" alt="dashboard" src="https://user-images.githubusercontent.com/5777880/180991553-cb4b2a19-5a69-45d8-924d-e7f8024ba770.png">
<img width="437" alt="plural" src="https://user-images.githubusercontent.com/5777880/180991559-db2eb855-fbec-46e9-9079-ad6615ee9e81.png">
<img width="390" alt="zero" src="https://user-images.githubusercontent.com/5777880/180991567-62f68200-2180-4e38-957d-380b3d9a4687.png">
<img width="375" alt="singular" src="https://user-images.githubusercontent.com/5777880/180991571-df793b90-a2c6-47a9-8821-7c99383d0558.png">
<img width="822" alt="table" src="https://user-images.githubusercontent.com/5777880/180991575-1d952232-b566-45f2-b86d-7feb663e678d.png">


I'm gonna close this issue now. If you need further help please feel free to reopen it along with an example repo with your code.

**Comment by NoTuxNoBux** [comment]
Thanks for the response. I figured out what the problem was. I got confused because the definition of `qrReferencePoints` in `User` (in your example) is what determines the column label, not the plural of the actual entity it refers to. This means I had to add a `label`, e.g. `User { qrReferencePoints: relationship({ label: 'The label' }) }`, to set it appropriately - because I didn't, it was using a default name based off the field name `qrReferencePoints`, which was what resulted in `Qr Reference Points`.

So, not a bug, just me being confused :sweat_smile: .
